clients/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceAfterImageConsumerImpl.java

```diff
@@ -97,6 +119,19 @@ public Collection<PubSubMessage<K, ChangeEvent<V>, Long>> poll(long timeoutInMs)
   // TODO: Find a better way to avoid data gap between version topic and change capture topic due to log compaction.
   private boolean handleControlMessage(ControlMessage controlMessage, PubSubTopicPartition pubSubTopicPartition) {
     ControlMessageType controlMessageType = ControlMessageType.valueOf(controlMessage);
+    if (controlMessageType.equals(ControlMessageType.START_OF_PUSH)) {
+      StartOfPush startOfPush = (StartOfPush) controlMessage.controlMessageUnion;
+      byte[] dictionary = null;
+      if (startOfPush.compressionDictionary != null) {
+        dictionary = startOfPush.compressionDictionary.array();
+      }
+      // TODO: This relies on consuming the beginning of the version topic. This is what some libraries do anyway under
+      // the hood, but it seems clumsy here. Should refactor
+      currentCompressor = compressorFactory.createVersionSpecificCompressorIfNotExist(
+          CompressionStrategy.valueOf(startOfPush.compressionStrategy),
+          pubSubTopicPartition.getPubSubTopic().getName(),
+          dictionary);
```

Comment:
If the`startOfPush` does not have compressionDictionary, then we will pass null to `compressorFactory.createVersionSpecificCompressorIfNotExist`, which might cause NPE there.

Reply:
I suppose so, but we should fail loudly.  That NPE would imply that we're using ZSTD compression but with no dictionary provided (and we only support ZSTD with dictionary today).  I think we should add that protection in the library itself.

Reply:
Added check in the library and unit test.